### DISCUSSION TITLE
Feature to transform Request Collection to Design Document

### DISCUSSION THREAD

**Discussion by pplotka** [Kong/insomnia#3833] [Ideas]
A built-in feature to do this is nowhere to be found, but sometimes (when you have a lot of requests) can be very useful.

**Comment by pplotka** [comment]
I wrote a blog post in which I described a workaround, but still, it will be fine to have this feature in the application: https://pplotka.dev/post/insomnia-transform-collection-to-document/

**Reply by MarkOSullivan94** [reply]
@pplotka big thank you for the instructions on how to convert collection to document! Found myself wanting to do the same this morning.

**Reply by Spider101** [reply]
@pplotka Thank you so much for the instructions. I was about to waste potentially hours of my time trying to manually convert the collection requests into the design document.

>Remove this project from Insomnia.

Does this mean that we cannot have a Collection and Design document with the same requests?

**Comment by dimitropoulos** [comment]
thanks @pplotka!  I can tell you that this is definitely on our minds to address.  The app was effectively forked at one point (so we could make insomnia designer) and then later re-merged back in, and what you're seeing is a remnant of that.  We want to fix this user experience.  Thank you for taking the time to write up your blog post.

**Comment by francoishill** [comment]
@pplotka, @MarkOSullivan94 and @Spider101 (cc @dimitropoulos).  Another way I found was to change the `scope` property in the `C:\Users\YOUR_USERNAME\AppData\Roaming\Insomnia\insomnia.Workspace.db` file (**I recommend to backup first** just in case).

Follow these steps:

* Close Insomnia application
* Backup your `C:\Users\YOUR_USERNAME\AppData\Roaming\Insomnia` folder (just in case)
* Change the relevant entry(ies) in `insomnia.Workspace.db`, change "scope":"collection"` to `"scope":"design"` or vice versa, each one is on a separate line in the file
* Open Insomnia and see the changes

**Comment by DannyHinshaw** [comment]
That'll be `$HOME/Library/Application\ Support/Insomnia/insomnia.Workspace.db` for the macos users.

[Relevant source code for any passers by](https://github.com/Kong/insomnia/blob/develop/packages/insomnia-inso/src/data-directory.ts#L9-L23).

Thank you!
